test/testdata/infer/proc_type_messages.rb

```diff
@@ -0,0 +1,49 @@
+# typed: true
+
+class ProcTypeErrorTest
+
+  extend T::Sig
+
+  sig {params(x: T.proc.params(x: Integer).returns(String)).void}
+  def foo(x)
+  end
+
+  sig {params(x: T.proc.params(x: T::Array[Integer]).returns(T::Array[String])).void}
+  def array_proc(x)
+  end
+
+  sig {params(x: T.proc.params(x: T::Hash[String, Integer]).returns(T::Hash[Integer, String])).void}
+  def hash_proc(x)
+  end
+
+  sig {void}
+  def test_proc_type_error
+    x = T.cast(nil, T.proc.params(x: String).returns(Integer))
+    foo(x)
+#       ^ error: Expected `T.proc.params(arg0: Integer).returns(String)` but found `T.proc.params(arg0: String).returns(Integer)` for argument `x`
+#       Detailed explanation:
+#         Integer is not a subtype of String for return type of Proc1
+#         String is not a supertype of Integer for Arg0 of Proc1
```

Comment:
It is not possible to use a `test/testdata/` test to make assertions about the `Detailed explanation`. Please consult the git history to see how other tests that have improved detailed explanations have tested their changes.